airtable.js

```diff
@@ -98,162 +96,126 @@ const toAirtableFormat = (record, table) => {
-        reject(err);
-        return;
-      }
-
-      resolve(fromAirtableFormat(record.fields, table));
+  return base(table)
+    .find(id)
+    .then(record => {
+      return fromAirtableFormat(record.fields, table);
+    })
+    .catch(err => {
+      throw err;
     });
-  });
 }
 
-// TODO: current implementation only returns the first page
 /*
   Given the desired table, field type (column), and field ('nick wong' or 'aivant@pppower.io'),
   return the associated record object.
 */
-function getRecordsByAttribute(table, fieldType, field) {
-  return new Promise(function(resolve, reject) {
-    base(table)
-      .select({
-        view: VIEW,
-        filterByFormula: `{${fieldType}}='${field}'`
-      })
-      .firstPage((err, records) => {
-        if (err) {
-          reject(err);
-          return;
-        }
-        if (!records || records.length < 1) {
-          console.log(`No record was retrieved using this ${fieldType}.`);
-          resolve([]);
-          // No need for this to throw an error, sometimes there's just no values
-          // reject(new Error(`No record was retrieved using this ${fieldType}.`));
-          return;
-        }
+function getRecordsByAttribute(table, fieldType, field, sort = []) {
+  return base(table)
+    .select({
+      view: VIEW,
+      filterByFormula: `{${fieldType}}='${field}'`,
+      sort
+    })
+    .all()
+    .then(records => {
+      if (!records || records.length < 1) {
+        console.log(`No record was retrieved using this ${fieldType}.`);
+        return [];
+        // No need for this to throw an error, sometimes there're just no values
+      }
 
-        resolve(
-          records.map(record => fromAirtableFormat(record.fields, table))
-        );
-      });
-  });
+      return records.map(record => fromAirtableFormat(record.fields, table));
+    })
+    .catch(err => {
+      throw err;
+    });
 }
 
 // Given a table and a record object, update a record on Airtable.
 function updateRecord(table, id, updatedRecord) {
-  return new Promise(function(resolve, reject) {
-    const transformedRecord = toAirtableFormat(updatedRecord, table);
-    base(table).update(
-      [
-        {
-          id,
-          fields: transformedRecord
-        }
-      ],
-      function(err, records) {
-        if (err) {
-          reject(err);
-          return;
-        }
-
-        const expectedLen = 1;
-        if (records.length !== expectedLen) {
-          reject(
-            new Error(
-              `${records.length} records returned from creating 1 record. Expected: ${expectedLen}`
-            )
-          );
-          return;
-        }
-
-        resolve(records[0].id); // TODO
+  const transformedRecord = toAirtableFormat(updatedRecord, table);
+  return base(table)
+    .update([
+      {
+        id,
+        fields: transformedRecord
       }
-    );
-  });
+    ])
+    .then(records => {
+      const expectedLen = 1;
+      if (records.length !== expectedLen) {
```

Comment:
yeah is there even a case where it returns more than 1? I guess the special case would be if it returned 0, but we could handle that better

Reply:
Yeah, hmm let me try it with a bad record ID and see what it does (or I could go look at the Airtable code, I guess)